File: examples/python/evm/account.quote_and_execute.py
# Usage: uv run python evm/account.quote_and_execute.py

import asyncio
from cdp import CdpClient
from dotenv import load_dotenv

load_dotenv()


async def main():
    async with CdpClient() as cdp:
        account = await cdp.evm.get_or_create_account(name="account")

        quote = await account.quote_fund(
            network="base",
            token="eth",
            amount=500000000000000, # 0.0005 eth
        )

        # get details of the quote
        print("Fiat amount: ", quote.fiat_amount)
        print("Fiat currency: ", quote.fiat_currency)
        print("Token amount: ", quote.token_amount)
        print("Token: ", quote.token)
        print("Network: ", quote.network)
        for fee in quote.fees:
            print("Fee type: ", fee.type)  # operation or network
            print("Fee amount: ", fee.amount)  # amount in the token
            print("Fee currency: ", fee.currency)  # currency of the amount

        response = await quote.execute()

        completed_transfer = await account.wait_for_fund_operation_receipt(
            transfer_id=response.id,
        )

        print(completed_transfer)


asyncio.run(main()) 

File: examples/typescript/evm/account.quoteAndExecute.ts
// Usage: pnpm tsx evm/account.quoteAndExecute.ts

import { CdpClient } from "@coinbase/cdp-sdk";
import "dotenv/config";

async function main() {
  const cdp = new CdpClient();
  const account = await cdp.evm.getOrCreateAccount({ name: "account" });

  const quote = await account.quoteFund({
    network: "base",
    token: "eth",
    amount: 500000000000000n, // 0.0005 eth
  });

  // get details of the quote
  console.log("Fiat amount: ", quote.fiatAmount)
  console.log("Fiat currency: ", quote.fiatCurrency)
  console.log("Token amount: ", quote.tokenAmount)
  console.log("Token: ", quote.token)
  console.log("Network: ", quote.network)
  for (const fee of quote.fees) {
    console.log("Fee type: ", fee.type) // network_fee or exchange_fee
    console.log("Fee amount: ", fee.amount) // amount in the token
    console.log("Fee currency: ", fee.currency) // currency of the amount
  }

  const response = await quote.execute();

  const completedTransfer = await account.waitForFundOperationReceipt({
    transferId: response.id
  });

  console.log(completedTransfer);
}

main().catch(console.error);
